lib/Dialect/VectorExt/Transform/VectorWarpUtils.cpp

```diff
@@ -720,15 +722,50 @@ static LogicalResult rewriteWarpOpToScfFor(RewriterBase &rewriter,
 
   // Allocate a shared memory buffer for the given type.
   auto allocBuffer = [&](Type type) {
-    OpBuilder::InsertionGuard g(rewriter);
-    rewriter.setInsertionPoint(ifOp);
+    // Compute type of shared memory buffer.
+    MemRefType memrefType;
     if (auto vectorType = type.dyn_cast<VectorType>()) {
-      return rewriter.create<memref::AllocOp>(
-          loc,
-          MemRefType::get(vectorType.getShape(), vectorType.getElementType()));
+      memrefType = MemRefType::get(
+          vectorType.getShape(), vectorType.getElementType(), {},
+          kSharedMemorySpace);
     } else {
-      return rewriter.create<memref::AllocOp>(loc, MemRefType::get({1}, type));
+      memrefType = MemRefType::get({1}, type, {}, kSharedMemorySpace);
     }
+
+    // Get symbol table holding all shared memory globals.
+    ModuleOp moduleOp = warpOp->getParentOfType<ModuleOp>();
+    SymbolTable symbolTable(moduleOp);
+
+    // Create a pretty name. Make sure that the name is unique.
```

Comment:
Note: This is copied from BufferUtils.cpp. Could be a helper in MemrefOps.

Reply:
I'm pretty sure the names doesn't have to be unique at this point and we don't need all this. I would skip all this logic.
Look at how iree does it:
https://github.com/google/iree/blob/45fc889981a3b3b88bbc11066bcfcf92f91d7c77/iree/compiler/Codegen/LLVMGPU/ConvertToLLVM.cpp#L149

Reply:
You're right, `SymbolTable::insert` already makes the symbol name unique. Dropped that part.